.github/workflows/release.yml

```diff
@@ -0,0 +1,248 @@
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.7
+
+      - name: Cache Go deps
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Download frontend build
+        uses: actions/download-artifact@v3
+        with:
+          name: build
+          path: frontend/build       
+
+      - name: Build image     
+        run: make clean build 
+
+      - name: Push image
+        run: |
+          docker login -u ${{ secrets.QUAY_USER }} -p ${{ secrets.QUAY_PASSWORD }} quay.io
+        
+          make -e DOCKER_CLI_EXPERIMENTAL=enabled build-linux-multi-arch container-multi-arch-push-kiali-quay
+
+      - name: Configure git
+        run: |
+          git config user.email 'kiali-dev@googlegroups.com'
+        
+          git config user.name 'kiali-bot'
+
+      - name: Create tag 
+        if: ${{ needs.initialize.outputs.release_type != 'edge' }}
+        run: |
+          git add Makefile
+        
+          git commit -m "Release $RELEASE_VERSION"
+        
+          git push origin $(git rev-parse HEAD):refs/tags/$RELEASE_VERSION
+        
+      - name: Create release
+        run: |
+          if [[ $RELEASE_TYPE == *"snapshot"* ]]; then export PRERELEASE="true"; else export PRERELEASE="false"; fi
+
+          curl -H "Authorization: token ${{ secrets.GH_TOKEN }}" \
+            -H "Content-Type: application/json" \
+            -d "{\"name\": \"Kiali Operator $RELEASE_VERSION\", \"tag_name\": \"$RELEASE_VERSION\", \"prerelease\": $PRERELEASE}" \
+            -X POST $GH_RELEASE_URI        
+
+      - name: Create or update version branch      
+        if: ${{ needs.initialize.outputs.release_type != 'edge' && !contains(needs.initialize.outputs.release_type, 'snapshot') }}
+        run: git push origin $(git rev-parse HEAD):refs/heads/$BRANCH_VERSION
+
+      - name: Create a PR to prepare for next version
+        env:
+          BUILD_TAG: kiali-release-${{ github.run_number }}-main
+        if: ${{ needs.initialize.outputs.release_type == 'minor' }}
+        run: |
+          sed -i -r "s/^VERSION \?= (.*)/VERSION \?= $NEXT_VERSION-SNAPSHOT/" Makefile
+          
+          git add Makefile
+          
+          git commit -m "Prepare for next version"
+          
+          git push origin $(git rev-parse HEAD):refs/heads/$BUILD_TAG
+          
+          curl -H "Authorization: token ${{ secrets.GH_TOKEN }}" \
+            -H "Content-Type: application/json" \
+            -d "{\"title\": \"Prepare for next version\", \"body\": \"Please, merge to update version numbers and prepare for release $NEXT_VERSION.\", \"head\": \"kiali-bot:$BUILD_TAG\", \"base\": \"$RELEASE_BRANCH\"}" \
+            -X POST $GH_PULL_URI      
```

Comment:
Same here. I'd suggest to use an already available action, if possible.
Check the Market: https://github.com/marketplace?type=actions&query=create+pull+request+

Reply:
I can check actions available but I saw they are third party mantained and I thought it would be better to keep the good old curl.

Reply:
Yes, I think it is mostly 3rd parties.
Well, if you want to stick with something official. I can suggest using the `gh` CLI. See: https://cli.github.com/manual/gh_release_create.

AFAIK, it is available in the runner and invoking `gh` will result in a more readable line than using `curl`.

Reply:
Also, I think the command is wrong. I see `\"head\": \"kiali-bot:$BUILD_TAG\"` which means that this would create a PR using a branch in the kiali-bot account, rather than the branch that you just created.

Reply:
gh CLI sounds good, I will try it. Thanks!.

Reply:
Now that you mention, you're right, but worked for me, because I used "leandroberetta" instead of "kiali-bot", and that is me and my repo. For this PR, needs to be removed. Good catch Edgar!.

Reply:
I changed the curl for the gh cli and works fine. I also fix the head field issue.